### DISCUSSION TITLE
DependencyProperty causes crash

### DISCUSSION THREAD

**Discussion by vb2ae** [microsoft/WindowsAppSDK#3178] [Q&A] [answered]
I am trying to set or get a value of a dependency property in code and my app exists.

For example this works 

    public sealed partial class HomeView : UserControl
    {
        public HomeView()
        {
            this.InitializeComponent();
            var obj = this as DependencyObject;
            obj.SetValue(IsGeneratedProperty, false);
        }

        public static readonly DependencyProperty IsGeneratedProperty =
    DependencyPropertyHelper.RegisterAttached(
        "IsGenerated",
        typeof(bool),
        typeof(DependencyObject));
    }

if I try to set the dependency property from a dependency property in a dll my app crashes

        public HomeView()
        {
            this.InitializeComponent();
            var obj = this as DependencyObject;
            obj.SetValue(View.IsGeneratedProperty, false);
        }
        

I see this in the output window

The thread 0x3878 has exited with code 0 (0x0).
The program '[26584] Caliburn.Micro.WinAppSdk.Test.exe' has exited with code 3221226107 (0xc000027b).

Any suggestions on how to fix this?

**Comment by symbiogenesis** [comment, marked as answer]
Is the DependencyObject actually a View? Most references online to this error make it sound like a type resolution error.
